### PR TITLE
Revert #53174 (applying the delta in move and collide) + doc improvement

### PR DESCRIPTION
**Pull request by fabriceci**
See discussion in #53174

### CODE REVIEW

File: doc/classes/PhysicsBody2D.xml [hunk 1]
```diff
@@ -25,12 +25,12 @@
...
 		<method name="move_and_collide">
 			<return type="KinematicCollision2D" />
-			<argument index="0" name="linear_velocity" type="Vector2" />
+			<argument index="0" name="distance" type="Vector2" />
 			<argument index="1" name="test_only" type="bool" default="false" />
 			<argument index="2" name="safe_margin" type="float" default="0.08" />
 			<description>
-				Moves the body along the vector [code]linear_velocity[/code]. This method should be used in [method Node._physics_process] (or in a method called by [method Node._physics_process]), as it uses the physics step's [code]delta[/code] value automatically in calculations. Otherwise, the simulation will run at an incorrect speed.
-				The body will stop if it collides. Returns a [KinematicCollision2D], which contains information about the collision when stopped, or when touching another body along the motion.
+				Moves the body along the vector [code]distance[/code]. In order to be frame rate independent in `_physics_process` or `_process`, [code]distance[/code] should be computed using [code]delta[/code].
```
Comment (akien-mga) [inline doc/classes/PhysicsBody2D.xml:32]:
```suggestion
				Moves the body along the vector [code]distance[/code]. In order to be frame rate independent in [method Node._physics_process] or [method Node._process], [code]distance[/code] should be computed using [code]delta[/code].
```

Comment (akien-mga) [reply doc/classes/PhysicsBody2D.xml:32]: Same for the other occurrences. For the reference, <code>`foo`</code> doesn't work for inline code formatting in the XML markup, it should be <code>[code]foo[/code]</code>. But here it's better to refer to the actual methods with a hyperlink.

File: scene/2d/physics_body_2d.cpp [hunk 2]
```diff
@@ -55,10 +55,7 @@ PhysicsBody2D::~PhysicsBody2D() {
 }
 
 Ref<KinematicCollision2D> PhysicsBody2D::_move(const Vector2 &p_linear_velocity, bool p_test_only, real_t p_margin) {
```
Comment (akien-mga) [inline scene/2d/physics_body_2d.cpp:57]: Isn't the C++ argument name wrong? It's a distance and not a velocity here if I understand correctly, so it should be `p_distance` (same in 3D).

Comment (fabriceci) [reply scene/2d/physics_body_2d.cpp]: Yes, strangely the change was not made in the PR that I reverted (for the record the old name used was `motion`)

**Review by akien-mga — APPROVED** [review]
We discussed this on chat and in the linked PR and the consensus is indeed to revert this change.

There's still room for further improvement in the clarity/consistency of this API, with ideas raised such as renaming `move_and_collide` and/or `move_and_slide` to clarify that one is high level and one is lower level. But this would be a topic for a new proposal for further ~bikeshedding~ discussion.

**Comment by akien-mga** [comment]
Thanks!

**Comment by fire-forge** [comment]
Should this be done to `move_and_slide` as well? It doesn't make sense to have some functions automatically multiply delta and some not. I often forget which variables I should and shouldn't multiply by delta when writing code, so I sometimes end up applying delta twice which isn't good. Removing any automatic delta multiplication would make this much easier to use.

**Comment by aaronfranke** [comment]
@fire-forge In Godot 4.0, `move_and_slide` does not take any arguments.

**Comment by fire-forge** [comment]
@aaronfranke sorry, I wasn't aware of that. I've only used Godot 3's `move_and_slide` so I didn't know.

**Comment by fabriceci** [comment]
> @aaronfranke sorry, I wasn't aware of that. I've only used Godot 3's `move_and_slide` so I didn't know.

Plus, move and slide needs to know the delta, it's better to manage it internally (otherwise we would have to add a new parameter to pass it).

**Pull request merged**